### PR TITLE
Use current user as default publisher

### PR DESCRIPTION
**Pull request by rubenvanerk**
Fix for https://github.com/rainlab/blog-plugin/issues/463

### CODE REVIEW

File: models/Post.php [hunk 1]
```diff
@@ -112,12 +113,16 @@ class Post extends Model
      */
     public function filterFields($fields, $context = null)
     {
+        $user = BackendAuth::getUser();
+        
+        if ($context == 'create') {
+            $fields->user->value = $user->id;
```
Comment (rubenvanerk) [inline models/Post.php]: I also tried `$fields->user->defaults = $user->id;` but that didn't seem to work?

File: models/Post.php [hunk 2]
```diff
@@ -112,12 +113,16 @@ class Post extends Model
      */
     public function filterFields($fields, $context = null)
     {
+        $user = BackendAuth::getUser();
+        
+        if ($context == 'create') {
```
Comment (bennothommo) [inline models/Post.php]: @rubenvanerk Please make this an explicit check via `===`
```suggestion
        if ($context === 'create') {
```

File: models/Post.php [hunk 3]
```diff
@@ -104,6 +104,7 @@ class Post extends Model
     public $preview = null;
 
     /**
+     * Set value for the field user to current user in the create context
```
Comment (bennothommo) [inline models/Post.php]: @rubenvanerk Could you move this comment line and the one below inside the function just above where those changes take place, and change this docblock's comment to say `Update fields based on current user.`?

File: models/Post.php (hunk 1 above)
Comment (bennothommo) [reply models/Post.php]: @rubenvanerk This line is fine, however, can you make sure that this line does not override any selection made in the form?

Comment (LukeTowers) [reply models/Post.php]: @rubenvanerk it should be `$fields->user->default` not `defaults`. I don't think it's a good idea to change the value here, if you were going to change the value itself it should be done in beforeSave only if no other value has been selected.

Comment (rubenvanerk) [reply models/Post.php]: I think I tried `default` too but didn't work. When I dumped the field I saw `defaults` which was filled with the `default` attribute in the yaml file. That's why I tried `defaults`. 
Anyway, the beforeSave method, it won't be transparent to the user what value will be used, I think.

Comment (rubenvanerk) [reply models/Post.php]: I just checked and `$fields->user->default` does not work to set the default value in the form.
If there's a better way to do this I'd love to know. But I think the default value should be reflected in the form so the form doesn't deviate from what is actually saved to the database.

Comment (bennothommo) [reply models/Post.php]: @LukeTowers I agree with @rubenvanerk that it should be transparent that it's defaulting to the current user, however, @LukeTowers suggestion could be met by setting the `empty` value of the publisher dropdown to be "Current user", and then `beforeSave` could replace the empty value with the authenticated user ID.

Comment (rubenvanerk) [reply models/Post.php]: @bennothommo I just pushed a commit that fixes the issue and complies with @LukeTowers suggestion.

File: models/Post.php [hunk 4]
```diff
@@ -139,6 +139,10 @@ public function afterValidate()
 
     public function beforeSave()
     {
+        if(empty($this->user)) {
```
Comment (bennothommo) [inline models/Post.php]: @rubenvanerk one more: could you add a space between `if` and `(`?

**Comment by bennothommo** [comment]
@rubenvanerk Looks fine to me, tested and working correctly. @LukeTowers, do you have any further feedback?

**Comment by LukeTowers** [comment]
@rubenvanerk nope, LGTM! Good job @rubenvanerk!

**Comment by bennothommo** [comment]
Nice work @rubenvanerk!

**Pull request merged**